main/lsp/requests/document_symbol.cc

```diff
@@ -77,6 +79,27 @@ bool DocumentSymbolTask::isDelayable() const {
     return true;
 }
 
+bool isOwnerInTheSameFile(const core::GlobalState &gs, core::SymbolRef ref, core::FileRef fref) {
+    auto locs = ref.data(gs)->owner.data(gs)->locs();
+
+    for (auto loc : locs) {
```

Comment:
if you wanna get fancy you can probably do `absl::c_any_of(locs, [fref](auto &loc) { return loc.file() == fref; });` in a few places in this file